src/components/modals/CategoryFilterModal.vue

```diff
@@ -0,0 +1,154 @@
+                        <span class="tag">No categories selected</span>
+                    </span>
+                </div>
+            </div>
+            <hr/>
+            <div>
+                <input
+                    v-model="allowNsfw"
+                    id="nsfwCheckbox"
+                    class="is-checkradio has-background-color"
+                    type="checkbox"
+                    :class="[{'is-dark': !isDarkTheme, 'is-white': isDarkTheme}]"
+                >
+                <label for="nsfwCheckbox">Allow NSFW (potentially explicit) mods</label>
+            </div>
+            <br/>
+            <div>
+                <div v-for="(key, index) in categoryFilterValues" :key="`cat-filter-${key}-${index}`">
+                    <input
+                        name="categoryFilterCondition"
+                        type="radio"
+                        :id="`cat-filter-${key}-${index}`"
+                        :value=key
+                        :checked="index === 0 ? true : undefined" v-model="categoryFilterMode"
+                    />
+                    <label :for="`cat-filter-${key}-${index}`">
+                        <span class="margin-right margin-right--half-width" />
+                        {{ key }}
+                    </label>
+                </div>
+            </div>
+        </template>
+        <template v-slot:footer>
+            <button class="button is-info" @click="close">
+                Apply filters
+            </button>
+        </template>
+    </modal>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue } from "vue-property-decorator";
+
+import { Modal } from '../../components/all';
+import CategoryFilterMode from "../../model/enums/CategoryFilterMode";
+import GameManager from "../../model/game/GameManager";
+import ManagerSettings from "../../r2mm/manager/ManagerSettings";
+
+@Component({
+    components: { Modal }
+})
+export default class CategoryFilterModal extends Vue {
+    settings: ManagerSettings = new ManagerSettings();
+
+    @Prop({default: () => null})
+    private onClose!: () => Promise<void>;
+
+    get allowNsfw(): boolean {
+        return this.$store.state.modFilters.allowNsfw;
+    }
+
+    set allowNsfw(value: boolean) {
+        this.$store.commit("modFilters/setAllowNsfw", value);
+    }
+
+    get categoryFilterMode(): CategoryFilterMode {
+        return this.$store.state.modFilters.categoryFilterMode;
+    }
+
+    set categoryFilterMode(value: CategoryFilterMode) {
+        this.$store.commit("modFilters/setCategoryFilterMode", value);
+    }
+
+    get categoryFilterValues() {
+        return Object.values(CategoryFilterMode);
+    }
+
+    close() {
+        this.onClose();
+        this.$store.commit("closeCategoryFilterModal");
+    }
+
+    async created() {
+        this.settings = await ManagerSettings.getSingleton(GameManager.activeGame);
+    }
+
+    // TODO: The value doesn't get updated if the theme is toggled while
```

Comment:
Might be good to link this TODO to something in clickup

Reply:
Removed this comment entirely since I don't think there's any way to toggle the theme while this modal is open, but if you feel like this should be tracked as a TODO let me know

Reply:
@MythicManiac While the modal isn't open, it's still mounted when user changes the color theme via settings, meaning the value won't get updated until user leaves and returns to the manager page. (And to clarify, the problem existed and was in fact worse before the changes in this PR.)

Also noticed you removed another TODO comment. Is this to say we should avoid TODO comments in this code base?